src/components/Trade/TradeInput.tsx

```diff
@@ -556,6 +601,39 @@ export const TradeInput = () => {
     [isSwapperApiPending, isSellAction],
   )
 
+  const ManualReceiveAddressEntry: JSX.Element = useMemo(() => {
+    return (
+      <FormControl>
+        <FormLabel color='gray.500' w='full'>
+          {translate('trade.receiveAddress')}
+        </FormLabel>
+        <AddressInput
+          showQr={false}
+          rules={{
+            required: true,
+            validate: {
+              validateAddress: async (rawInput: string) => {
+                const value = rawInput.trim() // trim leading/trailing spaces
+                setIsManualAddressEntryValidating(true)
+                const { assetId, chainId } = buyAsset
+                // this does not throw, everything inside is handled
+                const parseAddressInputArgs = { assetId, chainId, value }
+                const { address } = await parseAddressInput(parseAddressInputArgs)
```

Comment:
This is fine for now and realistically not a security risk for user funds, though I'm not sure we want this, since `parseAddressInput` handles:

1. addresses (which we obviously want to parse)
2. nslookups (which we want to support as well, and are able to get the address back from the domain name)
3. URLs - that one is the one I'm not sure we want here? We should theoretically be able to decode the address without a problem, and have tests for these utils, but it scares me knowing there is URL support, knowing all the ramifications WRT URLs, especially in the EVM world.

I reckon we should eventually make sure *not* to support 3. for safety reasons. The send modal has different steps, asset selection built-in etc, and many places where the address is shown, but in the context of `<TradeInput />` I'd rather avoid it.

Reply:
Also, are we going to eventually expose the manual address input even if the asset *is* supported by the wallet? Unless that's the case, I don't see any flow in which we would support ENS, since Ethereum is effectively be supported by all wallets (except Keplr, which we explicitly disable trading for)

Reply:
On the first comment, agree - lets not even attempt to call `parseMaybeUrlByChainId()` here.

Disabled in https://github.com/shapeshift/web/pull/4479/commits/3e618b7785f75e4a76b6729ad07c166be1bcddad by adding a flag to our `parseAddressInput()` input args.

Re: the second comment, whilst this is a product question, I expect we will want to eventually allow manual address entry regardless of wallet support (I've seen mocks to this effect in the Advance Trade settings UI). For this reason I've kept the functionally working so we can easily turn it on at a later date.

Reply:
Sounds good to me. Much safer with the flag now!
And ty for the rationale about keeping the name service logic working for when we will need it.